hip_data_tools/google/common.py

```diff
@@ -0,0 +1,32 @@
+from attr import dataclass
+
+
+@dataclass
+class GoogleApiConnectionSettings:
+    """Encapsulates the Google API connection settings"""
+    key_file_path: str
+
+
+class GoogleApiConnectionManager:
+    """
+    Google API connection manager abstract class
+    """
+
+    def __init__(self, settings: GoogleApiConnectionSettings):
+        self.settings = settings
+        self.scope = ['https://spreadsheets.google.com/feeds',
+                      'https://www.googleapis.com/auth/drive']
+
+    def _credentials(self):
+        """
+        Get the credentials for a given google service
+        Returns (ServiceAccountCredentials): credentials object to authorize google services
+        """
+        return
```

Comment:
not implemented?

Reply:
This class is a kind of abstract class. I'll use 'pass' instead of 'return'

Reply:
use pass and also annotate your methods with `@abstractmethod`

Reply:
fixed